src/AzureAppConfigurationImpl.ts

```diff
@@ -168,11 +175,44 @@ export class AzureAppConfigurationImpl extends Map<string, any> implements Azure
      */
     async load() {
         await this.#loadSelectedAndWatchedKeyValues();
-
         // Mark all settings have loaded at startup.
         this.#isInitialLoadCompleted = true;
     }
 
+    /**
+     * Construct hierarchical data object from map.
+     */
+    constructConfigurationObject(options?: ConfigurationObjectConstructionOptions): Record<string, any> {
+        const separator = options?.separator ?? ".";
+
+        // construct hierarchical data object from map
+        const data: Record<string, any> = {};
+        for (const [key, value] of this.#configMap) {
+            const segments = key.split(separator);
+            let current = data;
+            // construct hierarchical data object along the path
+            for (let i = 0; i < segments.length - 1; i++) {
+                const segment = segments[i];
+                // undefined or empty string
+                if (!segment) {
+                    throw new Error(`invalid key: ${key}`);
+                }
+                // create path if not exist
+                if (current[segment] === undefined) {
+                    current[segment] = {};
+                }
+                // The path has been occupied by a non-object value, causing ambiguity.
+                if (typeof current[segment] !== "object") {
+                    throw new Error(`The key '${segments.slice(0, i + 1).join(separator)}' is not a valid path.`);

```

Comment:
I think we can make some tweak on the error message here. Instead of saying it is not a valid path, we can say there is ambiguity.

Reply:
How about this:
```ts
new Error(`There is ambiguity constructing configuration object from key '${key}', value '${value}'.`)
```

Reply:
> How about this:
> 
> ```ts
> new Error(`There is ambiguity constructing configuration object from key '${key}', value '${value}'.`)
> ```

How about
``` ts
new Error(`There is ambiguity constructing configuration object from key '${key}', value '${value}'. The path '${segments.slice(0, i + 1).join(separator)}' has been occupied.`)
```

Reply:
your suggestion is better, providing more info about the ambiguity. BTW I would change it to `Ambiguity occurs when constructing .... `. 

Then for key-values:
- "app.settings.font", 1
- "app.settings.font.color", "red"

the error would be like:
> Ambiguity occurs when constructing configuration object from key 'app.settings.font.color', value 'red'. The path 'app.settings.font' has been occupied.

However, if I switch the order as you mentioned, the ambiguity check would be at the end, there seems no **trivial** way to provide clue about the occupied path.
I.e. key-values:
- "app.settings.font.color", "red"
- "app.settings.font", 1
> Ambiguity occurs when constructing configuration object from key 'app.settings.font', value '1'.

Reply:
In this case, I can add "The key should not be part of another key." And the error looks like:
> Ambiguity occurs when constructing configuration object from key 'app.settings.font', value '1'. The key should not be part of another key.